core/gaspool.go

```diff
@@ -25,6 +25,15 @@ import (
 // in a block. The zero value is a pool with zero gas available.
 type GasPool uint64
 
+// SetGas set an initial value for gaspool
+func (gp *GasPool) SetGas(amount uint64) *GasPool {
+	if amount > math.MaxUint64 {
+		panic("gas pool pushed above uint64")
```

Comment:
uint64 > math.MaxUint64, Is the check necessary?

Reply:
removed